### DISCUSSION TITLE
connmark not inststaling rules in iptables

### DISCUSSION THREAD

**Discussion by 0rlik** [strongswan/strongswan#788] [Q&A]
HI, what is missing in configuration ? 
#ipsec.conf
conn l2tp
	type=transport
	mark=%unique
	authby=secret
	pfs=no
	rekey=no
	keyingtries=1
	left=%any
	leftprotoport=udp/l2tp
	leftid=XX.XX.XX.XX
	right=%any
	rightprotoport=udp/%any
	auto=add

#connmark.conf
connmark {
    load = yes
}

After user is connecting I do not see any rules added to iptables mangle table 

ipsec statusall
Status of IKE charon daemon (strongSwan 5.9.4, Linux 5.14.0-4-amd64, x86_64):
  uptime: 9 minutes, since Nov 25 17:48:01 2021
  malloc: sbrk 2838528, mmap 0, used 679616, free 2158912
  worker threads: 11 of 16 idle, 5/0/0/0 working, job queue: 0/0/0/0, scheduled: 0
  loaded plugins: charon des aesni aes rc2 sha2 sha1 md5 mgf1 random nonce x509 revocation constraints pubkey pkcs1 pkcs7 pkcs8 pkcs12 pgp dnskey sshkey pem openssl fips-prf gmp agent xcbc hmac gcm drbg attr kernel-netlink resolve socket-default connmark forecast farp stroke updown eap-identity eap-aka eap-md5 eap-gtc eap-mschapv2 eap-radius eap-tls eap-ttls eap-tnc xauth-generic xauth-eap xauth-pam tnc-tnccs lookip error-notify certexpire led addrblock unity counters
Listening IP addresses:
  XX.XX.XX.XX
Connections:
        l2tp:  %any...%any  IKEv1/2
        l2tp:   local:  [xx.xx.xx.xx] uses pre-shared key authentication
        l2tp:   remote: uses pre-shared key authentication
        l2tp:   child:  dynamic[udp/l2f] === dynamic[udp] TRANSPORT
Security Associations (0 up, 0 connecting):

xt_connmark module loaded ..

**Comment by tobiasbrunner** [comment]
Your status output does not show any established SAs.

**Reply by 0rlik** [reply]
Here is with established session. mangle table is still empty

ipsec statusall
Status of IKE charon daemon (strongSwan 5.9.4, Linux 5.14.0-4-amd64, x86_64):
  uptime: 13 hours, since Nov 25 18:02:09 2021
  malloc: sbrk 2838528, mmap 0, used 851648, free 1986880
  worker threads: 11 of 16 idle, 5/0/0/0 working, job queue: 0/0/0/0, scheduled: 0
  loaded plugins: charon des aesni aes rc2 sha2 sha1 md5 mgf1 random nonce x509 revocation constraints pubkey pkcs1 pkcs7 pkcs8 pkcs12 pgp dnskey sshkey pem openssl fips-prf gmp agent xcbc hmac gcm drbg attr kernel-netlink resolve socket-default connmark forecast farp stroke updown eap-identity eap-aka eap-md5 eap-gtc eap-mschapv2 eap-radius eap-tls eap-ttls eap-tnc xauth-generic xauth-eap xauth-pam tnc-tnccs lookip error-notify certexpire led addrblock unity counters
Listening IP addresses:
  XX.XX.XX.XX

Connections:
        l2tp:  %any...%any  IKEv1/2
        l2tp:   local:  [XX.XX.XX.XX] uses pre-shared key authentication
        l2tp:   remote: uses pre-shared key authentication
        l2tp:   child:  dynamic[udp/l2f] === dynamic[udp] TRANSPORT
Security Associations (2 up, 0 connecting):
        l2tp[3]: ESTABLISHED 34 seconds ago, XX.XX.XX.XX[XX.XX.XX.XX]...2.132.157.122[192.168.0.110]
        l2tp[3]: IKEv1 SPIs: 45892433c79f67b0_i aff18e0543fac0ee_r*, rekeying disabled
        l2tp[3]: IKE proposal: AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048
        l2tp{4}:  INSTALLED, TRANSPORT, reqid 2, ESP in UDP SPIs: cb0d552f_i 005350d6_o
        l2tp{4}:  AES_CBC_256/HMAC_SHA2_256_128, 3464 bytes_i (42 pkts, 3s ago), 9036 bytes_o (45 pkts, 3s ago), rekeying disabled
        l2tp{4}:   XX.XX.XX.XX/32[udp/l2f] === 2.132.157.122/32[udp/51686]
        l2tp[2]: ESTABLISHED 9 hours ago, XX.XX.XX.XX[82.200.128.22]...109.206.9.55[192.168.0.10]
        l2tp[2]: IKEv1 SPIs: c2a373ca3645c546_i 6e1b2f41aaae5252_r*, rekeying disabled
        l2tp[2]: IKE proposal: AES_CBC_256/HMAC_SHA1_96/PRF_HMAC_SHA1/ECP_384
        l2tp{3}:  INSTALLED, TRANSPORT, reqid 1, ESP in UDP SPIs: c24abcea_i 5386468e_o
        l2tp{3}:  AES_CBC_256/HMAC_SHA1_96, 581513 bytes_i (4916 pkts, 29587s ago), 6747393 bytes_o (6928 pkts, 29587s ago), rekeying disabled
        l2tp{3}:  XX.XX.XX.XX/32[udp/l2f] === 109.206.9.55/32[udp/l2f]

**Reply by tobiasbrunner** [reply]
Read the log.

**Reply by 0rlik** [reply]
here is full log , start of strongswan and connecting user. I donot see any ifno regarding connmark. 
Do you have any example how this should be seen in log ? Too few info on strongswan.org and other sites in internet regarding connmark plugin

Nov 27 21:59:47 server systemd[1]: Started strongSwan IPsec IKEv1/IKEv2 daemon using ipsec.conf.
Nov 27 21:59:47 server ipsec[32384]: Starting strongSwan 5.9.4 IPsec [starter]...
Nov 27 21:59:47 server ipsec_starter[32384]: Starting strongSwan 5.9.4 IPsec [starter]...
Nov 27 21:59:47 server ipsec[32384]: # deprecated keyword 'pfs' in conn 'l2tp'
Nov 27 21:59:47 server ipsec[32384]:   PFS is enabled by specifying a DH group in the 'esp' cipher suite
Nov 27 21:59:47 server ipsec_starter[32384]: # deprecated keyword 'pfs' in conn 'l2tp'
Nov 27 21:59:47 server ipsec[32384]: ### 1 parsing error (0 fatal) ###
Nov 27 21:59:47 server ipsec_starter[32384]:   PFS is enabled by specifying a DH group in the 'esp' cipher suite
Nov 27 21:59:47 server ipsec_starter[32384]: ### 1 parsing error (0 fatal) ###
Nov 27 21:59:47 server charon[32388]: 00[DMN] Starting IKE charon daemon (strongSwan 5.9.4, Linux 5.14.0-4-amd64, x86_64)
Nov 27 21:59:47 server charon[32388]: 00[NET] using forecast interface eth0
Nov 27 21:59:47 server charon[32388]: 00[CFG] joining forecast multicast groups: 224.0.0.1,224.0.0.22,224.0.0.251,224.0.0.252,239.255.255.250
Nov 27 21:59:47 server charon[32388]: 00[CFG] loading ca certificates from '/etc/ipsec.d/cacerts'
Nov 27 21:59:47 server charon[32388]: 00[CFG]   loaded ca certificate "C=II, ST=ZZZZ, O=XXX, CN=EDED" from '/etc/ipsec.d/cacerts/strongswanCert.pem'
Nov 27 21:59:47 server charon[32388]: 00[CFG] loading aa certificates from '/etc/ipsec.d/aacerts'
Nov 27 21:59:47 server charon[32388]: 00[CFG] loading ocsp signer certificates from '/etc/ipsec.d/ocspcerts'
Nov 27 21:59:47 server charon[32388]: 00[CFG] loading attribute certificates from '/etc/ipsec.d/acerts'
Nov 27 21:59:47 server charon[32388]: 00[CFG] loading crls from '/etc/ipsec.d/crls'
Nov 27 21:59:47 server charon[32388]: 00[CFG] loading secrets from '/etc/ipsec.secrets'
Nov 27 21:59:47 server charon[32388]: 00[CFG]   loaded IKE secret for XX.XX.XX.XX %any
Nov 27 21:59:47 server charon[32388]: 00[CFG] loaded 0 RADIUS server configurations
Nov 27 21:59:47 server charon[32388]: 00[CFG] HA config misses local/remote address
Nov 27 21:59:47 server charon[32388]: 00[LIB] dropped capabilities, running as uid 0, gid 0
Nov 27 21:59:47 server charon[32388]: 00[JOB] spawning 16 worker threads
Nov 27 21:59:47 server ipsec[32384]: charon (32388) started after 20 ms
Nov 27 21:59:47 server ipsec_starter[32384]: charon (32388) started after 20 ms
Nov 27 21:59:47 server charon[32388]: 05[CFG] received stroke: add connection 'l2tp'
Nov 27 21:59:47 server charon[32388]: 05[CFG] added configuration 'l2tp'
:
Nov 27 21:59:53 server charon[32388]: 14[NET] received packet: from 2.132.157.122[500] to XX.XX.XX.XX[500] (788 bytes)
Nov 27 21:59:53 server charon[32388]: 14[ENC] parsed ID_PROT request 0 [ SA V V V V V V V V V V V V ]
Nov 27 21:59:53 server charon[32388]: 14[IKE] received NAT-T (RFC 3947) vendor ID
Nov 27 21:59:53 server charon[32388]: 14[IKE] received draft-ietf-ipsec-nat-t-ike vendor ID
Nov 27 21:59:53 server charon[32388]: 14[IKE] received draft-ietf-ipsec-nat-t-ike-08 vendor ID
Nov 27 21:59:53 server charon[32388]: 14[IKE] received draft-ietf-ipsec-nat-t-ike-07 vendor ID
Nov 27 21:59:53 server charon[32388]: 14[IKE] received draft-ietf-ipsec-nat-t-ike-06 vendor ID
Nov 27 21:59:53 server charon[32388]: 14[IKE] received draft-ietf-ipsec-nat-t-ike-05 vendor ID
Nov 27 21:59:53 server charon[32388]: 14[IKE] received draft-ietf-ipsec-nat-t-ike-04 vendor ID
Nov 27 21:59:53 server charon[32388]: 14[IKE] received draft-ietf-ipsec-nat-t-ike-03 vendor ID
Nov 27 21:59:53 server charon[32388]: 14[IKE] received draft-ietf-ipsec-nat-t-ike-02 vendor ID
Nov 27 21:59:53 server charon[32388]: 14[IKE] received draft-ietf-ipsec-nat-t-ike-02\n vendor ID
Nov 27 21:59:53 server charon[32388]: 14[IKE] received FRAGMENTATION vendor ID
Nov 27 21:59:53 server charon[32388]: 14[IKE] received DPD vendor ID
Nov 27 21:59:53 server charon[32388]: 14[IKE] 2.132.157.122 is initiating a Main Mode IKE_SA
Nov 27 21:59:53 server charon[32388]: 14[IKE] 2.132.157.122 is initiating a Main Mode IKE_SA
Nov 27 21:59:53 server charon[32388]: 14[CFG] selected proposal: IKE:AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048
Nov 27 21:59:53 server charon[32388]: 14[ENC] generating ID_PROT response 0 [ SA V V V V ]
Nov 27 21:59:53 server charon[32388]: 14[NET] sending packet: from XX.XX.XX.XX[500] to 2.132.157.122[500] (160 bytes)
Nov 27 21:59:53 server charon[32388]: 15[NET] received packet: from 2.132.157.122[500] to XX.XX.XX.XX[500] (380 bytes)
Nov 27 21:59:53 server charon[32388]: 15[ENC] parsed ID_PROT request 0 [ KE No NAT-D NAT-D ]
Nov 27 21:59:53 server charon[32388]: 15[IKE] remote host is behind NAT
Nov 27 21:59:53 server charon[32388]: 15[ENC] generating ID_PROT response 0 [ KE No NAT-D NAT-D ]
Nov 27 21:59:53 server charon[32388]: 15[NET] sending packet: from XX.XX.XX.XX[500] to 2.132.157.122[500] (396 bytes)
Nov 27 21:59:53 server charon[32388]: 01[NET] received packet: from 2.132.157.122[4500] to XX.XX.XX.XX[4500] (108 bytes)
Nov 27 21:59:53 server charon[32388]: 01[ENC] parsed ID_PROT request 0 [ ID HASH N(INITIAL_CONTACT) ]
Nov 27 21:59:53 server charon[32388]: 01[CFG] looking for pre-shared key peer configs matching XX.XX.XX.XX...2.132.157.122[192.168.0.107]
Nov 27 21:59:53 server charon[32388]: 01[CFG] selected peer config "l2tp"
Nov 27 21:59:53 server charon[32388]: 01[IKE] IKE_SA l2tp[1] established between XX.XX.XX.XX[XX.XX.XX.XX]...2.132.157.122[192.168.0.107]
Nov 27 21:59:53 server charon[32388]: 01[IKE] IKE_SA l2tp[1] established between XX.XX.XX.XX[XX.XX.XX.XX]...2.132.157.122[192.168.0.107]
Nov 27 21:59:53 server charon[32388]: 01[ENC] generating ID_PROT response 0 [ ID HASH ]
Nov 27 21:59:53 server charon[32388]: 01[NET] sending packet: from XX.XX.XX.XX[4500] to 2.132.157.122[4500] (92 bytes)
Nov 27 21:59:54 server charon[32388]: 09[NET] received packet: from 2.132.157.122[4500] to XX.XX.XX.XX[4500] (412 bytes)
Nov 27 21:59:54 server charon[32388]: 09[ENC] parsed QUICK_MODE request 1350094324 [ HASH SA No ID ID NAT-OA NAT-OA ]
Nov 27 21:59:54 server charon[32388]: 09[CFG] selected proposal: ESP:AES_CBC_256/HMAC_SHA2_256_128/NO_EXT_SEQ
Nov 27 21:59:54 server charon[32388]: 09[IKE] received 3600s lifetime, configured 0s
Nov 27 21:59:54 server charon[32388]: 09[ENC] generating QUICK_MODE response 1350094324 [ HASH SA No ID ID NAT-OA NAT-OA ]
Nov 27 21:59:54 server charon[32388]: 09[NET] sending packet: from XX.XX.XX.XX[4500] to 2.132.157.122[4500] (204 bytes)
Nov 27 21:59:54 server charon[32388]: 10[NET] received packet: from 2.132.157.122[4500] to XX.XX.XX.XX[4500] (76 bytes)
Nov 27 21:59:54 server charon[32388]: 10[ENC] parsed QUICK_MODE request 1350094324 [ HASH ]
Nov 27 21:59:54 server ipsec[32388]: 00[DMN] Starting IKE charon daemon (strongSwan 5.9.4, Linux 5.14.0-4-amd64, x86_64)
Nov 27 21:59:54 server ipsec[32388]: 00[NET] using forecast interface eth0
Nov 27 21:59:54 server ipsec[32388]: 00[CFG] joining forecast multicast groups: 224.0.0.1,224.0.0.22,224.0.0.251,224.0.0.252,239.255.255.250
Nov 27 21:59:54 server ipsec[32388]: 00[CFG] loading ca certificates from '/etc/ipsec.d/cacerts'
Nov 27 21:59:54 server ipsec[32388]: 00[CFG]   loaded ca certificate "C=II, ST=ZZZZ, O=XXX, CN=EDED" from '/etc/ipsec.d/cacerts/strongswanCert.pem'
Nov 27 21:59:54 server ipsec[32388]: 00[CFG] loading aa certificates from '/etc/ipsec.d/aacerts'
Nov 27 21:59:54 server ipsec[32388]: 00[CFG] loading ocsp signer certificates from '/etc/ipsec.d/ocspcerts'
Nov 27 21:59:54 server ipsec[32388]: 00[CFG] loading attribute certificates from '/etc/ipsec.d/acerts'
Nov 27 21:59:54 server ipsec[32388]: 00[CFG] loading crls from '/etc/ipsec.d/crls'
Nov 27 21:59:54 server ipsec[32388]: 00[CFG] loading secrets from '/etc/ipsec.secrets'
Nov 27 21:59:54 server ipsec[32388]: 00[CFG]   loaded IKE secret for XX.XX.XX.XX %any
Nov 27 21:59:54 server ipsec[32388]: 00[CFG] loaded 0 RADIUS server configurations
Nov 27 21:59:54 server ipsec[32388]: 00[CFG] HA config misses local/remote address
Nov 27 21:59:54 server ipsec[32388]: 00[LIB] dropped capabilities, running as uid 0, gid 0
Nov 27 21:59:54 server ipsec[32388]: 00[JOB] spawning 16 worker threads
Nov 27 21:59:54 server ipsec[32388]: 05[CFG] received stroke: add connection 'l2tp'
Nov 27 21:59:54 server ipsec[32388]: 05[CFG] added configuration 'l2tp'
Nov 27 21:59:54 server ipsec[32388]: 14[NET] received packet: from 2.132.157.122[500] to XX.XX.XX.XX[500] (788 bytes)
Nov 27 21:59:54 server ipsec[32388]: 14[ENC] parsed ID_PROT request 0 [ SA V V V V V V V V V V V V ]
Nov 27 21:59:54 server ipsec[32388]: 14[IKE] received NAT-T (RFC 3947) vendor ID
Nov 27 21:59:54 server ipsec[32388]: 14[IKE] received draft-ietf-ipsec-nat-t-ike vendor ID
Nov 27 21:59:54 server ipsec[32388]: 14[IKE] received draft-ietf-ipsec-nat-t-ike-08 vendor ID
Nov 27 21:59:54 server ipsec[32388]: 14[IKE] received draft-ietf-ipsec-nat-t-ike-07 vendor ID
Nov 27 21:59:54 server ipsec[32388]: 14[IKE] received draft-ietf-ipsec-nat-t-ike-06 vendor ID
Nov 27 21:59:54 server ipsec[32388]: 14[IKE] received draft-ietf-ipsec-nat-t-ike-05 vendor ID
Nov 27 21:59:54 server ipsec[32388]: 14[IKE] received draft-ietf-ipsec-nat-t-ike-04 vendor ID
Nov 27 21:59:54 server charon[32388]: 10[IKE] CHILD_SA l2tp{1} established with SPIs c93213b1_i 09336842_o and TS XX.XX.XX.XX/32[udp/l2f] === 2.132.157.122/32[udp/53370]
Nov 27 21:59:54 server ipsec[32388]: 14[IKE] received draft-ietf-ipsec-nat-t-ike-03 vendor ID
Nov 27 21:59:54 server ipsec[32388]: 14[IKE] received draft-ietf-ipsec-nat-t-ike-02 vendor ID
Nov 27 21:59:54 server ipsec[32388]: 14[IKE] received draft-ietf-ipsec-nat-t-ike-02\n vendor ID
Nov 27 21:59:54 server ipsec[32388]: 14[IKE] received FRAGMENTATION vendor ID
Nov 27 21:59:54 server ipsec[32388]: 14[IKE] received DPD vendor ID
Nov 27 21:59:54 server ipsec[32388]: 14[IKE] 2.132.157.122 is initiating a Main Mode IKE_SA
Nov 27 21:59:54 server ipsec[32388]: 14[CFG] selected proposal: IKE:AES_CBC_256/HMAC_SHA2_256_128/PRF_HMAC_SHA2_256/MODP_2048
Nov 27 21:59:54 server ipsec[32388]: 14[ENC] generating ID_PROT response 0 [ SA V V V V ]
Nov 27 21:59:54 server ipsec[32388]: 14[NET] sending packet: from XX.XX.XX.XX[500] to 2.132.157.122[500] (160 bytes)
Nov 27 21:59:54 server ipsec[32388]: 15[NET] received packet: from 2.132.157.122[500] to XX.XX.XX.XX[500] (380 bytes)
Nov 27 21:59:54 server ipsec[32388]: 15[ENC] parsed ID_PROT request 0 [ KE No NAT-D NAT-D ]
Nov 27 21:59:54 server ipsec[32388]: 15[IKE] remote host is behind NAT
Nov 27 21:59:54 server ipsec[32388]: 15[ENC] generating ID_PROT response 0 [ KE No NAT-D NAT-D ]
Nov 27 21:59:54 server ipsec[32388]: 15[NET] sending packet: from XX.XX.XX.XX[500] to 2.132.157.122[500] (396 bytes)
Nov 27 21:59:54 server ipsec[32388]: 01[NET] received packet: from 2.132.157.122[4500] to XX.XX.XX.XX[4500] (108 bytes)
Nov 27 21:59:54 server ipsec[32388]: 01[ENC] parsed ID_PROT request 0 [ ID HASH N(INITIAL_CONTACT) ]
Nov 27 21:59:54 server ipsec[32388]: 01[CFG] looking for pre-shared key peer configs matching XX.XX.XX.XX...2.132.157.122[192.168.0.107]
Nov 27 21:59:54 server ipsec[32388]: 01[CFG] selected peer config "l2tp"
Nov 27 21:59:54 server ipsec[32388]: 01[IKE] IKE_SA l2tp[1] established between XX.XX.XX.XX[XX.XX.XX.XX]...2.132.157.122[192.168.0.107]
Nov 27 21:59:54 server ipsec[32388]: 01[ENC] generating ID_PROT response 0 [ ID HASH ]
Nov 27 21:59:54 server ipsec[32388]: 01[NET] sending packet: from XX.XX.XX.XX[4500] to 2.132.157.122[4500] (92 bytes)
Nov 27 21:59:54 server ipsec[32388]: 09[NET] received packet: from 2.132.157.122[4500] to XX.XX.XX.XX[4500] (412 bytes)
Nov 27 21:59:54 server ipsec[32388]: 09[ENC] parsed QUICK_MODE request 1350094324 [ HASH SA No ID ID NAT-OA NAT-OA ]
Nov 27 21:59:54 server ipsec[32388]: 09[CFG] selected proposal: ESP:AES_CBC_256/HMAC_SHA2_256_128/NO_EXT_SEQ
Nov 27 21:59:54 server ipsec[32388]: 09[IKE] received 3600s lifetime, configured 0s
Nov 27 21:59:54 server ipsec[32388]: 09[ENC] generating QUICK_MODE response 1350094324 [ HASH SA No ID ID NAT-OA NAT-OA ]
Nov 27 21:59:54 server ipsec[32388]: 09[NET] sending packet: from XX.XX.XX.XX[4500] to 2.132.157.122[4500] (204 bytes)
Nov 27 21:59:54 server ipsec[32388]: 10[NET] received packet: from 2.132.157.122[4500] to XX.XX.XX.XX[4500] (76 bytes)
Nov 27 21:59:54 server ipsec[32388]: 10[ENC] parsed QUICK_MODE request 1350094324 [ HASH ]
Nov 27 21:59:54 server charon[32388]: 10[IKE] CHILD_SA l2tp{1} established with SPIs c93213b1_i 09336842_o and TS XX.XX.XX.XX/32[udp/l2f] === 2.132.157.122/32[udp/53370]
Nov 27 21:59:54 server xl2tpd[1396]: Connection established to 2.132.157.122, 53370.  Local: 27278, Remote: 4 (ref=0/0).  LNS session is 'default'
Nov 27 21:59:54 server xl2tpd[1396]: start_pppd: I'm running:
Nov 27 21:59:54 server xl2tpd[1396]: "/usr/sbin/pppd"
Nov 27 21:59:54 server xl2tpd[1396]: "/dev/pts/2"
Nov 27 21:59:54 server xl2tpd[1396]: "passive"
Nov 27 21:59:54 server xl2tpd[1396]: "nodetach"
Nov 27 21:59:54 server xl2tpd[1396]: "172.17.17.65:172.17.17.66"
Nov 27 21:59:54 server xl2tpd[1396]: "auth"
Nov 27 21:59:54 server xl2tpd[1396]: "require-chap"
Nov 27 21:59:54 server xl2tpd[1396]: "file"
Nov 27 21:59:54 server xl2tpd[1396]: "/etc/ppp/options.xl2tpd"
Nov 27 21:59:54 server xl2tpd[1396]: Call established with 2.132.157.122, PID: 32410, Local: 55492, Remote: 7092, Serial: 1
Nov 27 21:59:54 server pppd[32410]: pppd 2.4.9 started by root, uid 0
Nov 27 21:59:54 server pppd[32410]: Using interface ppp0
Nov 27 21:59:54 server pppd[32410]: Connect: ppp0 <--> /dev/pts/2
Nov 27 21:59:54 server systemd-udevd[32411]: Using default interface naming scheme 'v249'.
Nov 27 21:59:54 server charon[32388]: 13[KNL] interface ppp0 activated
Nov 27 21:59:54 server pppd[32410]: local  LL address fe80::5cd1:49c4:564d:e8b8
Nov 27 21:59:54 server pppd[32410]: remote LL address fe80::c8b4:2b2a:6e54:3b76
Nov 27 21:59:54 server charon[32388]: 01[KNL] fe80::5cd1:49c4:564d:e8b8 appeared on ppp0
Nov 27 21:59:54 server charon[32388]: 06[KNL] flags changed for fe80::5cd1:49c4:564d:e8b8 on ppp0
Nov 27 21:59:54 server charon[32388]: 09[KNL] 172.17.17.65 appeared on ppp0
Nov 27 21:59:54 server charon[32388]: 11[KNL] 172.17.17.65 disappeared from ppp0
Nov 27 21:59:54 server charon[32388]: 13[KNL] 172.17.17.65 appeared on ppp0
Nov 27 21:59:54 server pppd[32410]: Cannot determine ethernet address for proxy ARP
Nov 27 21:59:54 server pppd[32410]: local  IP address 172.17.17.65
Nov 27 21:59:54 server pppd[32410]: remote IP address 172.17.17.66
Nov 27 21:59:54 server charon[32388]: 01[NET] using forecast interface eth0
Nov 27 21:59:54 server charon[32388]: 01[CFG] joining forecast multicast groups: 224.0.0.1,224.0.0.22,224.0.0.251,224.0.0.252,239.255.255.250

**Reply by tobiasbrunner** [reply]
The _connmark_ plugin doesn't do much logging, except for errors. Since none are logged and the plugin is loaded (according to the status output), that means rules should have been installed for each CHILD_SA that was established (note that xl2tpd would not have received any packets if that wasn't the case as packets would not be marked correctly to match the inbound policy), see [connmark](https://wiki.strongswan.org/projects/strongswan/wiki/connmark) for details.

**Reply by 0rlik** [reply]
yeah , but i do not see any rules installed. And second client from same NAT ip  can not connect to this vpn.

**Reply by tobiasbrunner** [reply]
Please post the output of `ip xfrm state`, `ip xfrm policy`, and `iptables-save` after a client connected.

**Reply by tobiasbrunner** [reply]
Read the last line (it's even a warning).

**Reply by 0rlik** [reply]
Yep, but it is still empty
```
# iptables-legacy-save
# Generated by iptables-save v1.8.7 on Mon Nov 29 20:45:00 2021
*mangle
:PREROUTING ACCEPT [1760:275346]
:INPUT ACCEPT [1760:275346]
:FORWARD ACCEPT [0:0]
:OUTPUT ACCEPT [1649:265809]
:POSTROUTING ACCEPT [1649:265809]
COMMIT
# Completed on Mon Nov 29 20:45:00 2021
```

**Reply by tobiasbrunner** [reply]
That's with a client connected?

**Reply by 0rlik** [reply]
Sorry Tobias, server is dead, i will try to reconnect client, collect logs , and will return in 1-2 days ...

**Reply by tobiasbrunner** [reply]
Don't know anything about `xl2tpd`, so can't really help you with this. But depending on the client behavior, you might have to refer to the last section of [connmark](https://wiki.strongswan.org/projects/strongswan/wiki/connmark#Windows-L2TP) (i.e. use a patched version of `xl2tpd`).

**Reply by 0rlik** [reply]
looks like xl2tp can not differentiate two users connected from same NAT ip

**Comment by 0rlik** [comment]
Is it possible to enable debugging for this module ?

**Reply by tobiasbrunner** [reply]
Not without code changes.
